iocore/net/UnixNetVConnection.cc

```diff
@@ -79,9 +79,12 @@ static inline int
 read_signal_and_update(int event, UnixNetVConnection *vc)
 {
   vc->recursion++;
-  if (vc->read.vio.cont) {
+  if (vc->read.vio.cont && vc->read.vio.mutex == vc->read.vio.cont->mutex) {
     vc->read.vio.cont->handleEvent(event, &vc->read.vio);
   } else {
+    if (vc->read.vio.cont) {
+      Note("read_signal_and_update: mutexes are different? vc=%p, event=%d", vc, event);
+    }
```

Comment:
I assume this is just for debugging, right?  There should be no legitimate reason for the mutexes to mismatch.

Reply:
Yeah, it's just for debugging. Haven't seen any instances of this log. We can probably even add release_assert here